<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Logger Example</title>
    <script type="text/javascript" src="dist/pino-logger-client.js"></script>
</head>
<body>

<input id="logValue" type="text" placeholder="Write here what you want to log" value="I'm going to log this">
<br />
<button onclick="logMessage('info')">Log Info</button>
<br />
<button onclick="logMessage('warn')">Log Warn</button>
<br />
<button onclick="logMessage('error')">Log Error</button>
<br />
<button onclick="logMessage('success')">Log Success</button>
<br />
<button onclick="throwGlobalError()">Throw Window Error</button>
<br />
<button onclick="generateRejectedPromise(true)">Throw Unhandled Rejection</button>
<br />
Open Console >

<script type="text/javascript">
    const consoleLogger = new pinoLoggerClient("http://localhost:3000", "Pino");
    consoleLogger.info('This is just a normal info / log message');
    consoleLogger.warn('A warn message has a yellow color and an attention icon');
    consoleLogger.error('This error log puts up a stacktrace of the message to the console');
    consoleLogger.success('Is just a log / info message, but a different color.');
    const logMessage = (cType) => {
        switch (cType) {
            case 'info':
                consoleLogger.info(document.querySelector('#logValue').value);
                break;
            case 'warn':
                consoleLogger.warn(document.querySelector('#logValue').value);
                break;
            case 'error':
                consoleLogger.error(document.querySelector('#logValue').value);
                break;
            case 'success':
                consoleLogger.success(document.querySelector('#logValue').value);
                break;
        }
    }

    const throwGlobalError = () => {
        JSON.parse('a');
    }

    const generateRejectedPromise = (isEventuallyHandled) => {
        // Create a promise which immediately rejects with a given reason.
        var rejectedPromise = Promise.reject({error: 'Error at ' +
                new Date().toLocaleTimeString()});

        if (isEventuallyHandled) {
            // We need to handle the rejection "after the fact" in order to trigger a
            // unhandledrejection followed by rejectionhandled. Here we simulate that
            // via a setTimeout(), but in a real-world system this might take place due
            // to, e.g., fetch()ing resources at startup and then handling any rejected
            // requests at some point later on.
            setTimeout(() => {
                // We need to provide an actual function to .catch() or else the promise
                // won't be considered handled.
                rejectedPromise.catch(() => {});
            }, 1);
        }
    }
</script>
</body>
</html>
